hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileSystem.java

```diff
@@ -795,15 +796,19 @@ public void testListStatusOnKeyNameContainDelimiter() throws Exception {
    *
    * @throws IOException DB failure
    */
-  protected void deleteRootDir() throws IOException, InterruptedException {
+  protected void deleteRootDir()
+      throws IOException, InterruptedException, TimeoutException {
     FileStatus[] fileStatuses = fs.listStatus(ROOT);
 
     if (fileStatuses == null) {
       return;
     }
     deleteRootRecursively(fileStatuses);
-    Thread.sleep(500);
     fileStatuses = fs.listStatus(ROOT);
+    FileStatus[] finalFileStatuses = fileStatuses;
+    GenericTestUtils.waitFor(
```

Comment:
You are supposed to fetch `fs.listStatus(ROOT);` inside the lambda.

https://github.com/apache/ozone/pull/5217#discussion_r1308026768